frontend/src/components/ProfilePage.js

```diff
@@ -0,0 +1,98 @@
+import {
+    Box,
+    Button,
+    TextField,
+} from "@mui/material";
+
+import {useState, useContext} from "react";
+import {Link} from "react-router-dom";
+import axios from "axios";
+import { URL_USER_SVC, LOGOUT, DASHBOARD} from "../configs";
+import {STATUS_CODE_CONFLICT, STATUS_CODE_SUCCESS} from "../constants";
+import { UserContext } from "../util/userContext";
+
+function ProfilePage() {
+    const {user,setUser} = useContext(UserContext)
+    const [oldPassword, setOldPassword] = useState("")
+    const [newPassword, setNewPassword] = useState("")
+    const [newPasswordRepeat, setNewPasswordRepeat] = useState("")
+    const handleLogout = async () => {
+        const accesstoken = user.accesstoken
+        const res = await axios.post(URL_USER_SVC+LOGOUT,{withCredentials:true,credentials: "include"},{headers: {
+            Authorization: 'Bearer ' + accesstoken
+        }})
+            .catch((err) => {
+                if (err.response.status === STATUS_CODE_CONFLICT) {
+                    console.log("Error during log out")
+                }
+            })
+        if (res && res.status === STATUS_CODE_SUCCESS) {
+            setUser(null)
+        }
+    }
+    const handleUpdate = async () => {
+        const accesstoken = user.accesstoken
+        axios.interceptors.request.use(
+            config => {
+              config.headers['Authorization'] = `Bearer ${accesstoken}`;
+                  return config;
+              },
+              error => {
+                  return Promise.reject(error);
+              }
+          );
+        const res = await axios.put(URL_USER_SVC, {oldPassword, newPassword },{withCredentials:true,credentials: "include"}).catch((err) => {
+                if (err.response.status === STATUS_CODE_CONFLICT) {
+                    console.log('Incorrect username or password')
+                } else {
+                    console.log('Please try again later')
+                }
+            })
+        if (res && res.status === STATUS_CODE_SUCCESS) {
+            const accesstoken  = res.data.accesstoken
```

Comment:
Is this accesstoken used somewhere?
Are we missing a return statement here?

Reply:
I think some aspect of this feature is incomplete? Because even though the password update is implemented correctly, the frontend is not receiving the response. 
As per my understanding:
- When the password is updated, the user should see some success message and be logged out to login again?
- When the password update is failed, the user should see some message describing the failure of the update. Currently the server hangs since the response is not sent.

Reply:
![image](https://user-images.githubusercontent.com/65403285/191162915-d9ec9971-091d-448f-b20b-60d42f702ec5.png)

![image](https://user-images.githubusercontent.com/65403285/191162998-52f4703a-1ec0-48a6-824b-80655339cb30.png)

Implemented the frontend to receive and react according to the server response!